addons/account/wizard/account_payment_register.py

```diff
@@ -417,16 +417,15 @@ def _compute_show_require_partner_bank(self):
     @api.depends('source_amount', 'source_amount_currency', 'source_currency_id', 'company_id', 'currency_id', 'payment_date')
     def _compute_amount(self):
         for wizard in self:
-            if wizard.source_currency_id == wizard.currency_id:
-                # Same currency.
-                wizard.amount = wizard.source_amount_currency
-            elif wizard.currency_id == wizard.company_id.currency_id:
-                # Payment expressed on the company's currency.
-                wizard.amount = wizard.source_amount
+            if wizard.source_currency_id:
+                wizard.amount = wizard.source_currency_id._convert(
+                    wizard.source_amount_currency,
+                    wizard.currency_id,
+                    wizard.company_id,
+                    wizard.payment_date,
+                )
             else:
-                # Foreign currency on payment different than the one set on the journal entries.
-                amount_payment_currency = wizard.company_id.currency_id._convert(wizard.source_amount, wizard.currency_id, wizard.company_id, wizard.payment_date)
-                wizard.amount = amount_payment_currency
+                wizard.amount = None
```

Comment:
If all invoices are sharing the same currency, the wizard is in edit mode with `source_currency_id` set. If not, the wizard is not in edit mode and you are not able to edit the `amount` field because you have no choice to create one payment for each currency with no partial

Reply:
ok. add a comment maybe?